src/axom/lumberjack/TextTagCombiner.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *******************************************************************************
+ * \file TextTagCombiner.hpp
+ *
+ * \brief This file contains the class implementation of the
+ * TextTagCombiner.
+ *******************************************************************************
+ */
+
+#ifndef TEXTTAGCOMBINER_HPP
+#define TEXTTAGCOMBINER_HPP
+
+#include "axom/lumberjack/Combiner.hpp"
+#include "axom/lumberjack/Message.hpp"
+
+#include <string>
+
+namespace axom
+{
+namespace lumberjack
+{
+/*!
+ *******************************************************************************
+ * \class TextTagCombiner
+ *
+ * \brief Combines Message classes if their Message::text and Message::tag
+ *        are equal.
+ *
+ *  This class can be added to Lumberjack's Lumberjack by calling
+ *  Lumberjack::addCombiner with a
+ *  TextTagCombiner instance as its parameter.
+ *
+ * \see Combiner Lumberjack
+ *******************************************************************************
+ */
+class TextTagCombiner : public Combiner
+{
+public:
+  TextTagCombiner() { }
+
+  /*!
+   *****************************************************************************
+   * \brief Returns the unique string identifier for this combiner. Used by
+   *  Lumberjack to differentiate between other combiners.
+   *****************************************************************************
+   */
+  const std::string id() { return m_id; }
+
+  /*!
+   *****************************************************************************
+   * \brief Function used by Lumberjack to indicate whether two messages should
+   *  be combined.
+   *
+   * They are not actually combined by this function. Message classes are
+   * triggered for combination if both Message::text and Message::tag are equal.
+   *
+   * \param [in] leftMessage One of the Messages to be compared.
+   * \param [in] rightMessage One of the Messages to be compared.
+   *****************************************************************************
+   */
+  bool shouldMessagesBeCombined(const Message& leftMessage,
+                                const Message& rightMessage)
+  {
+    return (leftMessage.text().compare(rightMessage.text()) == 0 &&
```

Comment:
Make this the default combiner because you would lose the tag in the current default

Reply:
`TextTagCombiner` is the new default for Lumberjack, [ReadTheDocs ](https://axom.readthedocs.io/en/feature-han12-slic_tags/axom/lumberjack/docs/sphinx/combiner_class.html)was also updated with the new combiner default.